state/relationunit.go

```diff
@@ -482,7 +483,26 @@ func (ru *RelationUnit) SettingsAddress() (network.Address, error) {
 	if crossmodel, err := ru.relation.IsCrossModel(); err != nil {
 		return network.Address{}, errors.Trace(err)
 	} else if !crossmodel {
-		return unit.PrivateAddress()
+		space, err := unit.GetSpaceForBinding("")
```

Comment:
I *think* the right value is:
ru.endpoint.Relation.Name
(it embeds, so it might also just be ru.endpoint.Name)